### DISCUSSION TITLE
Seeing cached data on useForm's defaultValue

### DISCUSSION THREAD

**Discussion by lifeiscontent** [edmundhung/conform#745] [Q&A]
is there a scenario where conform would cache defaultValue? I'm seeing some behavior where I will load some data, change some data on another page, and go back to the page where I originally loaded the data and seeing the previous data cached until I hard refresh

**Comment by lifeiscontent** [comment]
@edmundhung is there a reason why conform seems to cache the stale defaultValue even after a successful submission? I would assume the value handed back to conform from the lastResult would be used

**Reply by edmundhung** [reply]
The `initialValue` from the `lastResult` is used only when Conform is initializing a new form context. I never changes the defaultValue or other metadata like constraints until a form reset happens, or if the component is unmounted and so `useForm` initialize a new form context.

What is the exact issue you had?

**Reply by lifeiscontent** [reply]
I guess my expectation is that after an action has been run, useForm would update its state based off the `lastResult` from the action, it seems the only way to actually do this is within an effect, and calling `form.update`
